pentaho/src/main/java/pt/webdetails/cgg/CggService.java

```diff
@@ -83,6 +83,8 @@ public void draw(
 
     this.draw( script, type, outputType, attachmentName, multiChartOverflow, null,
       width, height, servletResponse, servletRequest, null );
+
+    return Response.status( servletResponse.getStatus() ).build();
```

Comment:
You need to guarantee that the servletResponse.getStatus() will return a valid status code, otherwise this will produce an error. For instance if the servletResponse.status haven't been yet properly initialized before.